internal/test/variables.go

```diff
@@ -165,7 +164,6 @@ func BuildDomainModel(orgID string) *model.Domain {
 		Type:                  pointy.Uint(model.DomainTypeIpa),
 		IpaDomain: &model.Ipa{
 			Model: gorm.Model{
-				ID:        1,
```

Comment:
I am having a look to it! It depends on the operation:
- For `INSERT` Model.ID would be expected to be 0 (but for `ipas` table, because it is a 1-1 relationship, so the `Model.ID` assigned for `domains` entity is the used for `ipas.Model.ID`; here normally the helper assign a return value of 1, just for testing purposes).
- For `UPDATE`, the `domains` and `ipas` use to provide the Model.ID as the moment of executing the operation, for domains it is known Model.ID value, `ipas` table follow the same rule above (it is `DELETED` but when the `INSERT` happens apply the above because of the 1-1 relationship to `domains` table).
- The Update repository action is implemented in a special way, because it was easier to delete from `ipas` table, and create the entry again with the new values, rather than update/insert/delete depending on the resulting `ipas`, `ipa_certs`, `ipa_servers`, `ipa_locations` entities.

An effort to update without `DELETE` the `ipas` entity, when updating, can be found at: https://github.com/podengo-project/idmsvc-backend/pull/41  but it was not maintained since the last change, and repo has moved forward since then. Wondering if it should be closed that pr; but the correctness is better than the current implementation (if we are updating, expected the ID keep going to be the same, so the updated information should be correlated with the current one, and update/insert/delete entries as needed to keep the ID).

---

During this change, it was observed the expectations for some operations didn't fit the real scenario, and they were updated to match it; in short, `INSERT` only includes `id` field for `ipas` entity that is the only that know the `id` to 
be used.

Additional changes in relation with `id` field were observed, and updated to make the code more rugged.

Reply:
:+1: Closing #41